cpp/oneapi/dal/io/detail/load_graph.hpp

```diff
@@ -53,126 +54,127 @@ edge_list<std::int32_t> load_edge_list(const std::string &name) {
 
 template <typename Graph>
 void convert_to_csr_impl(const edge_list<vertex_type<Graph>> &edges, Graph &g) {
-    auto layout = oneapi::dal::preview::detail::get_impl(g);
-    using vertex_t = typename Graph::vertex_type;
-    using vector_vertex_t = typename Graph::vertex_set;
-    using vector_edge_t = typename Graph::edge_set;
-    using allocator_t = typename Graph::allocator_type;
-    using vertex_size_t = typename Graph::vertex_size_type;
-
     if (edges.size() == 0) {
-        layout->_vertex_count = 0;
-        layout->_edge_count = 0;
         throw invalid_argument("Empty edge list");
     }
 
-    vertex_t max_node_id = edges[0].first;
+    using vertex_t = typename Graph::vertex_type;
+    using edge_t = typename Graph::edge_type;
+    using vector_vertex_t = typename Graph::vertex_set;
+    using atomic_t = typename daal::services::Atomic<vertex_t>;
+
+    vertex_t max_id = edges[0].first;
     for (auto u : edges) {
-        vertex_t max_id_in_edge = std::max(u.first, u.second);
-        max_node_id = std::max(max_node_id, max_id_in_edge);
+        vertex_t edge_max = std::max(u.first, u.second);
+        max_id = std::max(max_id, edge_max);
     }
 
-    vertex_t _unf_vertex_count = max_node_id + 1;
+    const vertex_t n_vertex = max_id + 1;
 
-    using atomic_t = typename daal::services::Atomic<vertex_t>;
-    using allocator_atomic_t =
-        typename std::allocator_traits<allocator_t>::template rebind_alloc<atomic_t>;
+    auto layout = oneapi::dal::preview::detail::get_impl(g);
+    auto &allocator = layout->_allocator;
+    layout->_vertex_count = n_vertex;
+
+    void *degrees_vec_void =
+        (void *)allocator.allocate(n_vertex * (sizeof(atomic_t) / sizeof(char)));
+    atomic_t *degrees_vec = new (degrees_vec_void) atomic_t[n_vertex];
 
-    auto *degrees_vec = new (std::nothrow)
-        oneapi::dal::preview::detail::graph_container<atomic_t, allocator_atomic_t>(
-            _unf_vertex_count);
     if (degrees_vec == nullptr) {
         throw bad_alloc();
     }
-    atomic_t *degrees_cv = degrees_vec->data();
+    atomic_t *degrees_cv = degrees_vec;
     if (degrees_cv == nullptr) {
         throw bad_alloc();
     }
 
-    threader_for(_unf_vertex_count, _unf_vertex_count, [&](vertex_t u) {
-        degrees_cv[u].set(0);
-    });
-
     threader_for(edges.size(), edges.size(), [&](vertex_t u) {
         degrees_cv[edges[u].first].inc();
         degrees_cv[edges[u].second].inc();
     });
 
-    auto *rows_vec = new (std::nothrow)
-        oneapi::dal::preview::detail::graph_container<atomic_t, allocator_atomic_t>(
-            _unf_vertex_count + 1);
+    void *rows_vec_void =
+        (void *)allocator.allocate((n_vertex + 1) * (sizeof(atomic_t) / sizeof(char)));
+    atomic_t *rows_vec = new (rows_vec_void) atomic_t[n_vertex + 1];
```

Comment:
the same q as above: why not to simpy say 
```cpp
auto rows_vec = reinterpret_cast<atomic_t*>(allocator.allocate((n_vertex + 1)*sizeof(atomic_t)));
```

Reply:
It's not the same code